CMakeLists.txt

```diff
@@ -119,44 +200,37 @@ if(VAMP_BUILD_PYTHON)
     src/impl/vamp/bindings/settings.cc
     ${CMAKE_CURRENT_BINARY_DIR}/python.cc
   )
-
   foreach(robot_name robot_struct IN ZIP_LISTS VAMP_ROBOT_MODULES VAMP_ROBOT_STRUCTS)
-    configure_file(
-      src/impl/vamp/bindings/robot.cc.in
-      ${CMAKE_CURRENT_BINARY_DIR}/${robot_name}.cc
-      @ONLY
-    )
+  configure_file(
+    src/impl/vamp/bindings/robot.cc.in
+    ${CMAKE_CURRENT_BINARY_DIR}/${robot_name}.cc
+    @ONLY
+  )
 
-    list(APPEND VAMP_EXT_SOURCES
-      ${CMAKE_CURRENT_BINARY_DIR}/${robot_name}.cc
-    )
+  list(APPEND VAMP_EXT_SOURCES
+    ${CMAKE_CURRENT_BINARY_DIR}/${robot_name}.cc
+  )
   endforeach()
 
   nanobind_add_module(_core_ext
     NB_STATIC
     STABLE_ABI
     NOMINSIZE
     ${VAMP_EXT_SOURCES}
-    ${VAMP_SOURCES}
   )
 
   target_include_directories(_core_ext
     SYSTEM PRIVATE
-    ${VAMP_INCLUDES}
+    ${VAMP_COMMON_INCLUDES}
```

Comment:
Linking against an interface target will provide the necessary includes if they've been properly added to the interface target.